osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -238,9 +240,14 @@ protected override void LoadComplete()
 
             SelectedMods.BindValueChanged(val =>
             {
+                modSettingChangeTracker?.Dispose();
+
                 updateMultiplier();
                 updateCustomisation(val);
                 updateFromExternalSelection();
+
+                modSettingChangeTracker = new ModSettingChangeTracker(val.NewValue);
+                modSettingChangeTracker.SettingChanged += _ => updateMultiplier();
```

Comment:
Should probably be around an `if (AllowCustomisation)` similar to the above optimisation. Concern is selecting all mods in the freemods overlay.